### DISCUSSION TITLE
How to send and/or approve transactions without using useSendTransaction hook

### DISCUSSION THREAD

**Discussion by maleekcodes** [wevm/wagmi#798] [Q&A]
I am building a dApp that involves me approving transactions before I send them to the contract. I've tried using

`const provider = useProvider();
  const library = new Web3(provider);
`

But this doesn't work whenever I initiate the contract and make a send request to the contract. Below is my code:

```
const library = new Web3(provider);
    window.currentasset = new library.eth.Contract(
      ERC20Abi,
      token_address
    );

  try {
      await window.currentasset.methods
        .approve(address, APPROVE_VALUE)
        .send({ from: account });
      setApproval(true);
      setApprovalLoading(false);
    } catch (error: any) {
      if (error.code === 4001) {
        console.log("You denied the transaction.");
      } else {
        console.log("An error occurred. Please try again.");
      }
      setApprovalLoading(false);
    }
```

is there a way I can make this happen with wagmi?

**Comment by davigp** [comment]
you can call 'approve' function using usePrepareContractWrite & useContractWrite (https://wagmi.sh/docs/prepare-hooks/usePrepareContractWrite)
